pkg/sources/filesystem/filesystem.go

```diff
@@ -229,3 +230,43 @@ func (s *Source) Enumerate(ctx context.Context, units chan<- sources.Enumeration
 	}
 	return nil
 }
+
+// ChunkUnit implements SourceUnitChunker interface.
+func (s *Source) ChunkUnit(ctx context.Context, unit sources.SourceUnit, reporter sources.ChunkReporter) error {
+	path := unit.SourceUnitID()
+	logger := ctx.Logger().WithValues("path", path)
+
+	cleanPath := filepath.Clean(path)
+	fileInfo, err := os.Stat(cleanPath)
+	if err != nil {
+		return reporter.ChunkErr(ctx, fmt.Errorf("unable to get file info: %w", err))
+	}
+
+	ch := make(chan *sources.Chunk)
+	go func() {
+		defer close(ch)
+		if fileInfo.IsDir() {
+			// TODO: Finer grain error tracking of individual chunks.
+			err = s.scanDir(ctx, cleanPath, ch)
```

Comment:
Does this mean that `err` is shared between the goroutine and the calling function? Is this effectively a discard?

Reply:
Oh I see, I think some of the `err` shadowing confused me.